backend/src/main/java/com/redhat/cloud/notifications/processors/email/aggregators/RhosakEmailAggregator.java

```diff
@@ -1,10 +1,110 @@
 package com.redhat.cloud.notifications.processors.email.aggregators;
 
 import com.redhat.cloud.notifications.models.EmailAggregation;
+import io.vertx.core.json.JsonObject;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
 
 public class RhosakEmailAggregator extends AbstractEmailPayloadAggregator {
+    private static final String UPGRADES = "upgrades";
+    private static final String SERVICE_DISRUPTIONS = "disruptions";
+    private static final String EVENT_TYPE = "event_type";
+    private static final String SCHEDULED_UPGRADE_EVENT_TYPE = "scheduled-upgrade";
+    private static final String SERVICE_DISRUPTION_EVENT_TYPE = "disruption";
+    private static final String CONTEXT = "context";
+    private static final String EVENTS = "events";
+    private static final String PAYLOAD = "events";
+
+    private static final String PAYLOAD_NAME = "name";
+    private static final String KAFKA_VERSION = "kafka_version";
+    private static final String SERVICE_DISRUPTION_IMPACT = "impacted_area";
+
+    // know service disruption impacts
+    private static final String PERFORMANCE = "performance";
+    private static final String LATENCY = "latency";
+    private static final String THROUGHPUT = "throughput";
+
+    public RhosakEmailAggregator() {
+        context.put(UPGRADES, new JsonObject());
+        context.put(SERVICE_DISRUPTIONS, new JsonObject());
+    }
+
     @Override
     void processEmailAggregation(EmailAggregation aggregation) {
-        // TODO after the payloads are determined, notably the scheduled upgrade payload
+        JsonObject aggregationPayload = aggregation.getPayload();
+        String eventType = aggregationPayload.getString(EVENT_TYPE);
+        JsonObject context = aggregationPayload.getJsonObject(CONTEXT);
```

Comment:
Nothing of interest is stored in the events?

Reply:
Yes,

Please see:
- https://github.com/RedHatInsights/notifications-backend/blob/7f6d533c78848c7ccadf5d2b2494a093df77fc4c/backend/src/main/java/com/redhat/cloud/notifications/processors/email/aggregators/RhosakEmailAggregator.java#L54
- https://github.com/RedHatInsights/notifications-backend/blob/7f6d533c78848c7ccadf5d2b2494a093df77fc4c/backend/src/main/java/com/redhat/cloud/notifications/processors/email/aggregators/RhosakEmailAggregator.java#L91